connectors/sources/gitlab/datasource.py

```diff
@@ -0,0 +1,1040 @@
+
+This connector fetches:
+- Projects (repositories)
+- Issues (using Work Items API)
+- Merge Requests (using legacy API)
+- Epics (using Work Items API, group-level, requires Premium/Ultimate tier)
+- Releases (project-level version releases with changelogs)
+- README files (.md, .rst, .txt)
+
+This connector uses GitLab's Work Items API for Issues and Epics. Merge Requests
+continue to use the legacy GraphQL API as they have not yet been migrated to Work Items.
+
+Work Items API Reference: https://docs.gitlab.com/ee/api/graphql/reference/#workitem
+"""
+
+from functools import partial
+from typing import Any, AsyncGenerator, Type, TypeVar
+
+from pydantic import BaseModel
+
+from connectors.source import BaseDataSource, ConfigurableFieldValueError
+from connectors.sources.gitlab.client import GitLabClient
+from connectors.sources.gitlab.models import (
+    GitLabDiscussion,
+    GitLabIssue,
+    GitLabLabel,
+    GitLabMergeRequest,
+    GitLabProject,
+    GitLabRelease,
+    GitLabUser,
+    GitLabWorkItem,
+    PaginatedList,
+    WorkItemType,
+    WorkItemWidgetAssignees,
+    WorkItemWidgetDescription,
+    WorkItemWidgetHierarchy,
+    WorkItemWidgetLabels,
+    WorkItemWidgetLinkedItems,
+    WorkItemWidgetNotes,
+)
+from connectors.utils import decode_base64_value
+
+DEFAULT_PAGE_SIZE = 100
+SUPPORTED_EXTENSION = [".md", ".rst", ".txt"]
+
+
+class GitLabDataSource(BaseDataSource):
+    """GitLab Cloud Data Source."""
+
+    name = "GitLab"
+    service_type = "gitlab"
+    advanced_rules_enabled = False
+    dls_enabled = False
+    incremental_sync_enabled = False
+
+    def __init__(self, configuration) -> None:
+        """Setup the connection to the GitLab instance.
+
+        Args:
+            configuration (DataSourceConfiguration): Instance of DataSourceConfiguration class.
+        """
+        super().__init__(configuration=configuration)
+        self.gitlab_client = GitLabClient(
+            token=self.configuration["token"],
+        )
+        self.configured_projects = self.configuration["projects"]
+
+    def _set_internal_logger(self) -> None:
+        self.gitlab_client.set_logger(self._logger)
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for GitLab.
+
+        Returns:
+            dict: Default configuration.
+        """
+        return {
+            "token": {
+                "label": "Personal Access Token",
+                "order": 1,
+                "sensitive": True,
+                "type": "str",
+                "tooltip": "GitLab Personal Access Token with api, read_api, and read_repository scopes.",
+            },
+            "projects": {
+                "display": "textarea",
+                "label": "List of projects",
+                "order": 2,
+                "tooltip": "List of project paths (e.g., 'group/project'). Use '*' to sync all accessible projects.",
```

Comment:
What would `*` return for the user - is it going to be all projects that user starred? That user has been added to?

Reply:
That does make sense!